From 5333286dae3771a0fe4e3f0567287450afc2bb29 Mon Sep 17 00:00:00 2001
From: shyjumon <shyjumon.n@intel.com>
Date: Fri, 22 Mar 2019 22:33:18 +0530
Subject: [PATCH] HACK: mmc: cqhci: Workaround clock-gating problem by forcing
 SSC1 to 0x10004

Temporary workaround.

Refer https://hsdes.intel.com/resource/1209627782

Will be fixed by BIOS.

Signed-off-by: Adrian Hunter <adrian.hunter@intel.com>
Signed-off-by: shyjumon <shyjumon.n@intel.com>
---
 drivers/mmc/host/cqhci.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/mmc/host/cqhci.c b/drivers/mmc/host/cqhci.c
index 159270e..bb1777f 100644
--- a/drivers/mmc/host/cqhci.c
+++ b/drivers/mmc/host/cqhci.c
@@ -39,6 +39,8 @@ struct cqhci_slot {
 #define CQHCI_HOST_OTHER	BIT(4)
 };
 
+#define CQHCI_FORCE_SSC1 0x10004	/* WA: clock-gating problem by forcing SSC1 to 0x10004*/
+
 static inline u8 *get_desc(struct cqhci_host *cq_host, u8 tag)
 {
 	return cq_host->desc_base + (tag * cq_host->slot_sz);
@@ -263,6 +265,8 @@ static void __cqhci_enable(struct cqhci_host *cq_host)
 	cqhci_writel(cq_host, upper_32_bits(cq_host->desc_dma_base),
 		     CQHCI_TDLBAU);
 
+	cqhci_writel(cq_host, CQHCI_FORCE_SSC1, CQHCI_SSC1);
+
 	cqhci_writel(cq_host, cq_host->rca, CQHCI_SSC2);
 
 	cqhci_set_irqs(cq_host, 0);
-- 
2.7.4

